Update configure.ac build script
Remove references to guile which is only used for tests and breaks the package due to needing masked guile-1*
sed command in ebuild removes references to these tests
Taken from git repo which has updated the package: https://gitlab.com/Grimler/libmatheval/-/blob/master/configure.ac

--- a/configure.ac
+++ b/configure.ac
@@ -21,10 +21,8 @@ dnl Process this file with autoconf to produce a configure script.
 AC_PREREQ(2.53)
 AC_INIT([libmatheval],[1.1.11],[asamardzic@gnu.org])
 AC_CONFIG_AUX_DIR([config])
-AC_CONFIG_MACRO_DIR([m4])
-AC_SUBST([ac_aux_dir])
-AM_INIT_AUTOMAKE([libmatheval], [1.1.11])
-AC_CONFIG_SRCDIR([configure.in])
+AC_CONFIG_MACRO_DIRS([m4])
+AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER([config.h])
 
 dnl Initialize the test suite.
@@ -44,7 +42,6 @@ AC_PROG_LIBTOOL
 AC_PROG_MAKE_SET
 AC_PROG_YACC
 AM_MISSING_PROG([AUTOM4TE], [autom4te])
-GUILE_FLAGS
 
 dnl Checks for libraries.
 
@@ -59,11 +56,5 @@ AC_TYPE_SIZE_T
 dnl Checks for library functions.
 AC_CHECK_FUNCS([bzero memset], [break])
 
-dnl Additional Guile feature checks.
-AC_CHECK_TYPE([scm_t_bits], [AC_DEFINE([HAVE_SCM_T_BITS], [1], [Define to 1 if you have the `scm_t_bits' type.])], [], [#include <libguile.h>])
-AC_CHECK_LIB([guile], [scm_c_define_gsubr], [AC_DEFINE([HAVE_SCM_C_DEFINE_GSUBR], [1], [Define to 1 if you have the `scm_c_define_gsubr' function.])], [], [$GUILE_LDFLAGS])
-AC_CHECK_LIB([guile], [scm_make_gsubr], [AC_DEFINE([HAVE_SCM_MAKE_GSUBR], [1], [Define to 1 if you have the `scm_make_gsubr' function.])], [], [$GUILE_LDFLAGS])
-AC_CHECK_LIB([guile], [scm_num2dbl], [AC_DEFINE([HAVE_SCM_NUM2DBL], [1], [Define to 1 if you have the `scm_num2dbl' function.])], [], [$GUILE_LDFLAGS])
-
 AC_CONFIG_FILES([Makefile doc/Makefile lib/Makefile])
 AC_OUTPUT(libmatheval.pc)
